src/librustc/diagnostics.rs

```diff
@@ -168,6 +168,11 @@ match x {
 ```
 "##,
 
+E0010: r##"
+The value of static and const variables must be known at compile time, and they
+live for the entire lifetime of a program. They can not use allocation.
```

Comment:
The word "allocation" already occurs in the message currently emitted for `E0010`. Rather than "They can not use allocation" (which will not enlighten someone who did not already know what the first message meant), I would say "They are not allowed to create `Box` values", maybe?)